src/efftester.ml

```diff
@@ -369,7 +419,7 @@ end)
 
 let rec normalize_eff t =
   match t with
-  | Typevar _ | Unit | Int | Float | Bool | String -> t
+  | Typevar _ | Unit | Int | Float | Bool | String | Option _ -> t
```

Comment:
I think that you should recurse on the `Option` parameter, just like in the `List` case.